lib/bandit/http1/adapter.ex

```diff
@@ -75,13 +75,19 @@ defmodule Bandit.HTTP1.Adapter do
         end
 
       {:ok, {:http_request, method, {:abs_path, path}, version}, rest} ->
-        version =
-          case version do
-            {1, 1} -> :"HTTP/1.1"
-            {1, 0} -> :"HTTP/1.0"
-          end
-
-        do_read_headers(%{req | buffer: rest, version: version}, :httph, headers, method, path)
+        case get_version(version) do
+          {:ok, version} ->
+            do_read_headers(
+              %{req | buffer: rest, version: version},
+              :httph,
+              headers,
+              method,
+              path
+            )
+
+          {:error, reason} ->
+            {:error, reason}
+        end
```

Comment:
```suggestion
        with {:ok, version} <- get_version(version) do
          do_read_headers(%{req | buffer: rest, version: version}, :httph, headers, method, path)
        end
```

Reply:
Nice golfing!

Reply:
Since this is already merged I'll bring this in on main (can't credit you though :() Unless you want to spin up a PR?

Reply:
I can do it. There is a bunch of the same `case` types we could do in the same module, now that I'm looking at it. So, for consistency I can change them all or we just don't touch it it all?

Reply:
Do em all!